skll/utilities/compute_eval_from_predictions.py

```diff
@@ -39,7 +115,33 @@ def compute_eval_from_predictions(examples_file, predictions_file,
     pred = {}
     with open(predictions_file) as pred_file:
         reader = csv.reader(pred_file, dialect=csv.excel_tab)
-        next(reader)  # skip header
+        header = next(reader)
+
+        # If there are more than two columns, assume column 0 contains the ids, and
+        # columns 1-n contain class probabilities. Convert them to a class prediction
+        # using the specified `method`.
+        if len(header) > 2:
+            classes = [c for c in header[1:] if c]
+            if prediction_method is None:
+                prediction_method = "highest"
+                logger.info("No prediction method specified. Using 'highest'.")
+            if prediction_method == 'expected_value':
+                try:
+                    classes = [int(c) for c in classes]
+                except ValueError as e:
+                    raise e
+            for row in reader:
+                probabilities = [safe_float(p) for p in row[1:]]
+                prediction = get_prediction_from_probabilities(classes,
+                                                               probabilities,
+                                                               prediction_method)
+                pred[row[0]] = safe_float(prediction)
+        else:
+            if prediction_method is not None:
+                logger.warning("A prediction method was provided, but the predictions "
+                               "file doesn't contain probabilities. Ignoring prediction "
+                               "method '{}'.".format(prediction_method))
+
         for row in reader:
```

Comment:
Is there an indentation error here? Shouldn't this be nested underneath the `else` as part of the block that gets the predictions when they're not probabilities?

Reply:
Good catch.